code/modules/mob/living/simple_animal/hostile/terror_spiders/procs.dm

```diff
@@ -0,0 +1,447 @@
+
+/mob/living/simple_animal/hostile/poison/terror_spider/proc/DoShowOrders(var/usecache = 0, var/thequeen = null, var/theempress = null)
+	if (ckey)
+		to_chat(src, "------------------------")
+		if (ai_type == TS_AI_AGGRESSIVE)
+			to_chat(src, "Your Orders: <span class='danger'>kill all humanoids on sight! </span>")
+		else if (ai_type == TS_AI_DEFENSIVE)
+			to_chat(src, "Your Orders: <span class='notice'>defend yourself & the hive, without being aggressive </span> ")
+		else if (ai_type == TS_AI_PASSIVE)
+			to_chat(src, "Your Orders: <span class='danger'>do not attack anyone, not even in self-defense!</span> ")
+		to_chat(src, "------------------------")
+
+/mob/living/simple_animal/hostile/poison/terror_spider/proc/BroadcastOrders()
+	var/cache_thequeen = null
+	var/cache_theempress = null
+	for(var/mob/living/simple_animal/hostile/poison/terror_spider/queen/Q in ts_spiderlist)
+		if (Q.ckey || !Q.spider_awaymission)
+			cache_thequeen = Q
+			break
+	for(var/mob/living/simple_animal/hostile/poison/terror_spider/empress/E in ts_spiderlist)
+		if (E.ckey || !E.spider_awaymission)
+			cache_theempress = E
+			break
+	for(var/mob/living/simple_animal/hostile/poison/terror_spider/S in ts_spiderlist)
+		if (S.stat != DEAD)
+			S.DoShowOrders(1,cache_thequeen,cache_theempress)
+
+/mob/living/simple_animal/hostile/poison/terror_spider/proc/IsInfected(var/mob/living/B)
+	if (iscarbon(B))
+		var/mob/living/carbon/C = B
+		if (C.get_int_organ(/obj/item/organ/internal/body_egg))
+			return 1
+	return 0
+
+
+/mob/living/simple_animal/hostile/poison/terror_spider/proc/Retaliate()
+	var/list/around = view(src, 7)
+	var/list/ts_nearby = list()
+	for(var/atom/movable/A in around)
+		if (A == src)
+			continue
+		if (A in enemies)
+			continue
+		if (istype(A, /mob/living/simple_animal/hostile/poison/terror_spider))
+			ts_nearby += A
+			continue
+		if (isliving(A))
+			var/mob/living/M = A
+			var/faction_check = 0
+			for(var/F in faction)
+				if (F in M.faction)
+					faction_check = 1
+					break
+			if (faction_check && attack_same || !faction_check)
+				enemies |= M
+				visible_message("<span class='danger'>\icon[src] [src] glares at [M]! </span>")
+				// should probably exempt people who are dead...
+		else if (istype(A, /obj/mecha))
+			var/obj/mecha/M = A
+			if (M.occupant)
+				enemies |= M
+				enemies |= M.occupant
+		else if (istype(A, /obj/spacepod))
+			var/obj/spacepod/M = A
+			if (M.occupant || M.occupant2)
+				enemies |= M
+				enemies |= M.occupant
```

Comment:
@Kyep This is actually failing to compile - <a href=https://github.com/ParadiseSS13/Paradise/blob/master/code/modules/spacepods/spacepod.dm>the spacepod code's changed</a>, and `occupant` is no longer defined on them.